components/lock.rst

```diff
@@ -0,0 +1,257 @@
+the ``createLock()`` method. If needed, these locks can also be released early
+with the ``release()`` method.
+
+The trickiest part when working with expiring locks is choosing the right TTL.
+If it's too short, other processes could acquire the lock before finishing your
+work; it it's too long and the process crashes before calling the ``release()``
+method, the resource will stay locked until the timeout::
+
+    // ...
+    // create a expiring lock that lasts 30 seconds
+    $lock = $factory->createLock('charts-generation', 30);
+
+    $lock->acquire();
+    try {
+        // perform a job during less than 30 seconds
+    } finally {
+        $lock->release();
+    }
+
+.. tip::
+
+    To avoid letting the lock in a locking state, it's recommended to wrap the
+    job in a try/catch/finally block to always try to release the expiring lock.
+
+In case of long-running tasks, it's better to start with a not too long TTL and
+then use the :method:`Symfony\\Component\\Lock\\LockInterface::refresh` method
+to reset the TTL to its original value::
+
+    // ...
+    $lock = $factory->createLock('charts-generation', 30);
+
+    $lock->acquire();
+    try {
+        while (!$finished) {
+            // perform a small part of the job.
+
+            // renew the lock for 30 more seconds.
+            $lock->refresh();
+        }
+    } finally {
+        $lock->release();
+    }
+
+Available Stores
+----------------
+
+Locks are created and managed in ``Stores``, which are classes that implement
+:class:`Symfony\\Component\\Lock\\StoreInterface`. The component includes the
+following built-in store types:
+
+
+============================================  ======  ========  ========
+Store                                         Scope   Blocking  Expiring
+============================================  ======  ========  ========
+:ref:`FlockStore <lock-store-flock>`          local   yes       no
+:ref:`MemcachedStore <lock-store-memcached>`  remote  no        yes
+:ref:`RedisStore <lock-store-redis>`          remote  no        yes
+:ref:`SemaphoreStore <lock-store-semaphore>`  local   yes       no
+============================================  ======  ========  ========
+
+.. _lock-store-flock:
+
+FlockStore
+~~~~~~~~~~
+
+The FlockStore uses the file system on the local computer to create the locks.
+It does not support expiration, but the lock is automatically released when the
+PHP process is terminated::
+
+    use Symfony\Component\Lock\Store\FlockStore;
+
+    // the argument is the path of the directory where the locks are created
+    $store = new FlockStore(sys_get_temp_dir());
+
+.. caution::
+
+    Beware that some file systems (such as some types of NFS) do not support
+    locking. In those cases, it's better to use a local file or a remote store
+    based on Redis or Memcached.
+
+.. _lock-store-memcached:
+
+MemcachedStore
+~~~~~~~~~~~~~~
+
+The MemcachedStore saves locks on a Memcached server, so first you must create
```

Comment:
"so first you must create a Memcached connection implements the ``\Memcached`` class."
=>
"it requires a Memcached connection implementing the ``\Memcached`` class."